Source/Core/DolphinNoGUI/CMakeLists.txt

```diff
@@ -13,6 +13,11 @@ if(WIN32)
   target_sources(dolphin-nogui PRIVATE PlatformWin32.cpp)
 endif()
 
+if(APPLE)
+  target_sources(dolphin-nogui PRIVATE PlatformMacos.mm)
+  target_compile_options(dolphin-nogui PRIVATE -fno-objc-arc)
```

Comment:
Since it looks like you're not doing manual retain/release, I think you want `-fobjc-arc` here